File: app/src/main/java/com/thales/github/ui/repository/RepositoryContract.java
package com.thales.github.ui.repository;

import com.thales.github.model.repository.Repository;
import com.thales.github.ui.BasePresenter;

import java.util.List;

public interface RepositoryContract {
    interface View {
        void showRepositories(List<Repository> repositories);

        void showProgress();

        void hideProgress();

        void showSnackbarNoConnection();

        void showSnackbarError();

        void showNoRepositoriesMessage();

        void hideNoRepositoriesMessage();

        boolean isNetworkAvailable();

        void showProgressList();

        void hideProgressList();

        void showPullRequestUi(Repository repository);
    }

    interface Presenter extends BasePresenter {
        void loadRepositories();

        void setCurrentPage(int currentPage);

        void openPullRequest(Repository repository);
    }
}

File: app/src/main/java/com/thales/github/ui/repository/RepositoryContract.kt
package com.thales.github.ui.repository

import com.thales.github.model.repository.Repository
import com.thales.github.ui.BasePresenter

interface RepositoryContract {
    interface View {
        fun showRepositories(repositories: List<Repository>)

        fun showProgress()

        fun hideProgress()

        fun showSnackbarNoConnection()

        fun showSnackbarError()

        fun showNoRepositoriesMessage()

        fun hideNoRepositoriesMessage()

        fun isNetworkAvailable(): Boolean

        fun showProgressList()

        fun hideProgressList()

        fun showPullRequestUi(repository: Repository)
    }

    interface Presenter : BasePresenter {
        fun loadRepositories()

        fun setCurrentPage(currentPage: Int)

        fun openPullRequest(repository: Repository)
    }
}
